applications/StructuralMechanicsApplication/custom_processes/assign_nodal_elements_to_nodes_process.cpp

```diff
@@ -0,0 +1,303 @@
+        auto it_elem = it_elem_begin + i;
+        it_elem->SetId(i + 1);
+    });
+
+    // We get the reference element
+    const bool rayleigh_damping = mThisParameters["rayleigh_damping"].GetBool();
+    if (rayleigh_damping) {
+        p_properties->SetValue(CONSIDER_RAYLEIGH_DAMPING, true);
+    }
+    PointerVector<Node> aux_node_array(1);
+    const auto it_node_begin = mrThisModelPart.NodesBegin();
+    aux_node_array(0) = *(it_node_begin).base();
+
+    const SizeType number_of_nodes = mrThisModelPart.Nodes().size();
+
+    GeometryType::Pointer p_dummy_geom = GetPointGeometryFromNode(aux_node_array, domain_size);
+    const Element& r_reference_element = NodalConcentratedElement(0, p_dummy_geom);
+
+    std::vector<Element::Pointer> auxiliary_elements_vector  = IndexPartition<std::size_t>(number_of_nodes).for_each<AccumReduction<Element::Pointer>>([&](std::size_t i) {
+        auto it_node = it_node_begin + i;
+
+        PointerVector<Node> this_node_array(1);
+        this_node_array(0) = *(it_node).base();
+
+        auto p_element = r_reference_element.Create(number_elements + 1 + i, GetPointGeometryFromNode(this_node_array, domain_size), p_properties);
+
+        // Deep copy elemental flags
+        p_element->Set(Flags(*it_node));
+
+        return p_element;
+    });
+
+    // Adding to the model part
+    ElementsArrayType aux_elems;
+    aux_elems.GetContainer() = auxiliary_elements_vector;
+    mrThisModelPart.AddElements(aux_elems.begin(), aux_elems.end());
+
+    // We Initialize the elements
+    EntitiesUtilities::InitializeEntities<Element>(mrThisModelPart);
+
+    // Inactive by default
+    VariableUtils().SetFlag(ACTIVE, false, mrThisModelPart.Elements());
+
+    // Set the flag ACTIVE
+    this->Set(ACTIVE, false);
+
+    KRATOS_CATCH("")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void AssignNodalElementsToNodesProcess::ExecuteInitializeSolutionStep()
+{
+    KRATOS_TRY
+
+    // Check the interval
+    if (mThisParameters["interval"][0].GetDouble() > 0.0 || mThisParameters["interval"][1].GetDouble() < 1e30) {
+        if (this->IsNot(ACTIVE)) {
+            // Initialize initial displacement and rotation
+            block_for_each(mrThisModelPart.Elements(), [&](Element& rElement) {
+                if (rElement.Has(NODAL_INITIAL_DISPLACEMENT)) {
+                    rElement.SetValue(NODAL_INITIAL_DISPLACEMENT, rElement.GetGeometry()[0].FastGetSolutionStepValue(DISPLACEMENT));
+                }
+                if (rElement.Has(NODAL_INITIAL_ROTATION)) {
+                    rElement.SetValue(NODAL_INITIAL_ROTATION, rElement.GetGeometry()[0].FastGetSolutionStepValue(ROTATION));
+                }
+            });
+            // Set the flag ACTIVE
+            VariableUtils().SetFlag(ACTIVE, true, mrThisModelPart.Elements());
```

Comment:
Shouldn't this be done once at the `Initialize`? My point is that in here you might be overwriting the active/inactive state set by the user. Am I wrong?